File: qe-selenium-java/src/main/java/demos/ActivityTwelve.java
package demos;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.time.Duration;

public class ActivityTwelve {
    public static void main(String[] args) {

        WebDriver driver = new FirefoxDriver();
        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(10));

        driver.get("https://training-support.net/webelements/dynamic-content");
        System.out.println(driver.getTitle());

        WebElement genButton = driver.findElement(By.id("genButton"));
        genButton.click();

        WebElement word = driver.findElement(By.id("word"));

        wait.until(ExpectedConditions.textToBePresentInElement(word, "release"));
        System.out.println("Word is: " + word.getText());

        driver.quit();
    }
}

File: qe-selenium-python/Activity_Twelve.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC

driver = webdriver.Firefox()
wait = WebDriverWait(driver, 10)

driver.get("https://training-support.net/webelements/dynamic-content")
print(driver.title)

gen_button = driver.find_element(By.ID, "genButton")
gen_button.click()

word = driver.find_element(By.ID, "word")

wait.until(EC.text_to_be_present_in_element(word, "release"))
print("Word is:", word.text)

driver.quit()
